src/dotnet/commands/dotnet-msbuild/MSBuildForwardingApp.cs

```diff
@@ -36,7 +46,16 @@ public MSBuildForwardingApp(IEnumerable<string> argsToForward)
 
         public int Execute()
         {
-            return _forwardingApp.Execute();
+            try
+            {
+                Environment.SetEnvironmentVariable(TelemetrySessionIdEnvironmentVariableName, Telemetry.CurrentSessionId);
```

Comment:
It's really just there in case someone executes another instance of `dotnet` under this instance.